src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -1502,18 +1502,19 @@ private bool ExactOrBoundsNullableInference(ExactOrBoundsKind kind, TypeSymbolWi
                 return true;
             }
 
-            if (isNullableOnly(source) && isNullableOnly(target))
+            if (s_isNullableOnly(source) && s_isNullableOnly(target))
             {
                 ExactOrBoundsInference(kind, source.AsNotNullableReferenceType(), target.AsNotNullableReferenceType(), ref useSiteDiagnostics);
                 return true;
             }
 
             return false;
-
-            // True if the type is nullable but not an unconstrained type parameter.
-            bool isNullableOnly(TypeSymbolWithAnnotations type) => type.NullableAnnotation.IsAnyNullable() && !type.TypeSymbol.IsTypeParameterDisallowingAnnotation();
         }
 
+        // True if the type is nullable but not an unconstrained type parameter.
+        private readonly static Func<TypeSymbolWithAnnotations, bool> s_isNullableOnly =
+            (type) => type.NullableAnnotation.IsAnyNullable() && !type.TypeSymbol.IsTypeParameterDisallowingAnnotation();
```

Comment:
Is there a reason to use a delegate rather than a static method?

Reply:
Sorry about that. I was fixing a couple of places that allocate delegates and got trigger happy on this one which does not have a delegate. 
Reverted